ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -869,6 +869,37 @@ object Trees {
      */
     final case class Global(path: List[String]) extends JSNativeLoadSpec
 
+    /** Load from a module import.
+     *
+     *  The `module` is the ES module identifier. The `path` is a series of
+     *  nested property names starting from the module object.
+     *
+     *  The path can be empty, in which case the specification denotes the
+     *  module itself (whose exports are available as fields).
+     *
+     *  Any element in the path is a property selection from there. A module
+     *  import info with one path element is importing that particular value
+     *  from the module.
+     *
+     *  Examples:
+     *  {{{
+     *  // import { Bar as x } from 'foo'
+     *  Import("foo", List("Bar"))
+     *
+     *  // import { Bar as y } from 'foo'
+     *  // y.Baz
+     *  Import("foo", List("Bar", "Baz"))
+     *
+     *  // import * as x from 'foo' (whole module import)
+     *  Import("foo", Nil)
+     *
+     *  // import x from 'foo' (default import)
+     *  Import("foo", List("default"))
```

Comment:
The situation here is extremely unclear, wrt. interactions between ES2015 modules and "legacy" modules. There is at least [one well-regarded source](https://hacks.mozilla.org/2015/08/es6-in-depth-modules/) (section Default exports) saying that a legacy module would appear to ES6 as a module with a default import. Which would mean that your two lines above (CommonJS and ES2015) are actually equivalent if `my_lib.js` is a legacy module.

The [example for "external imports" from Rollup](http://rollupjs.org/?version=0.36.1&shareable=JTdCJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyJTJGJTJGJTIwWW91JTIwY2FuJTIwaW1wb3J0JTIwZXh0ZXJuYWwlMjBtb2R1bGVzJTIwaW50byUyMHlvdXIlMjBidW5kbGUlMjAlRTIlODAlOTMlNUNuJTJGJTJGJTIwaXQlMjBkb2Vzbid0JTIwbWF0dGVyJTIwaWYlMjB0aGV5J3JlJTIwRVMyMDE1JTIwb3IlMjBsZWdhY3klNUNuaW1wb3J0JTIwJTI0JTIwZnJvbSUyMCdqcXVlcnknJTNCJTVDbiU1Q24lMjQoJTIwJ2JvZHknJTIwKS5odG1sKCUyMCclM0NoMSUzRUhlbGxvJTIwd29ybGQhJTNDJTJGaDElM0UnJTIwKSUzQiUyMiU3RCU1RCU3RA==) seems to somewhat corroborate this view, as they translate the following ES2015 module:
```js
// You can import external modules into your bundle –
// it doesn't matter if they're ES2015 or legacy
import $ from 'jquery';

$( 'body' ).html( '<h1>Hello world!</h1>' );
```
into the following, with a magical `_interopDefault` to fallback to the module itself if it doesn't have a `"default"` field:
```js
function _interopDefault (ex) {
  return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex;
}
var $ = _interopDefault(require('jquery'));
$( 'body' ).html( '<h1>Hello world!</h1>' );
```
Note that none of this is *specified* per se. It all depends on the module Loader that orchestrates imports. Loaders can be different on different platforms (Node.js vs browser, for example), and for neither does there exist a spec at this point.

So I think it is fair to say, with the current understanding of ES modules, that the current specification for `Import` is likely going to do what one wants in practice. It certainly would with the current implementation of Rollup.